redis/asyncio/client.py

```diff
@@ -741,13 +751,13 @@ async def reset(self):
             self.patterns = {}
             self.pending_unsubscribe_patterns = set()
 
-    def close(self) -> Awaitable[NoReturn]:
-        # In case a connection property does not yet exist
-        # (due to a crash earlier in the Redis() constructor), return
-        # immediately as there is nothing to clean-up.
-        if not hasattr(self, "connection"):
-            return
-        return self.reset()
+    async def close(self) -> None:
```

Comment:
Good catch. This never should have returned Awaitable[NoReturn].

For the curious readers [see this](https://mypy.readthedocs.io/en/latest/more_types.html). Note that only a function that unconditionally raises an exception should do so.